tests/structures/test_class.py

```diff
@@ -46,6 +46,23 @@ def __str__(self):
             print('Done.')
             """, run_in_function=False)
 
+    # @unittest.expectedFailure
+    def test_getattr(self):
+        self.assertCodeExecution("""
+            class MyClass:
+                def __init__(self):
+                    self.x = 1
+                    self.y = 22
+                def __getattr__(self, attr):
+                    self.y = 42
+                    return y+1
+
+            obj = MyClass()
+            print(obj.x)
+            print(obj.fail)
+            print(obj.y)
+        """)
+
```

Comment:
I think that, in view of the current implementation and that this is Javascript, you should also add tests in the spirit of my negative example above (setting `__getattr__` and `__getattribute__` on the instance to verify that it doesn't work).